CodeEdit/Features/Settings/Pages/GeneralSettings/GeneralSettingsView.swift

```diff
@@ -0,0 +1,356 @@
+//
+//  GeneralSettingsView.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 4/1/23.
+//
+
+import SwiftUI
+
+/// A view that implements the `General` settings page
+struct GeneralSettingsView: View {
+    private let inputWidth: Double = 160
+    private let textEditorWidth: Double = 220
+    private let textEditorHeight: Double = 30
+
+    @EnvironmentObject
+    var updater: SoftwareUpdater
+
+    @AppSettings var settings
+
+    @State
+    private var openInCodeEdit: Bool = true
+
+    init() {
+        guard let defaults = UserDefaults.init(
+            suiteName: "austincondiff.CodeEdit.shared"
+        ) else {
+            print("Failed to get/init shared defaults")
+            return
+        }
+
+        self.openInCodeEdit = defaults.bool(forKey: "enableOpenInCE")
+    }
+
+    var body: some View {
+        SettingsForm {
+            Section {
+                appearance
+                fileIconStyle
+                tabBarStyle
+            }
+            Section {
+                showIssues
+                showLiveIssues
+            }
+            Section {
+                autoSave
+                revealFileOnFocusChangeToggle
+                reopenBehavior
+                afterWindowsCloseBehaviour
+                fileExtensions
+            }
+            Section {
+                projectNavigatorSize
+                findNavigatorDetail
+                issueNavigatorDetail
+            }
+            Section {
+                openInCodeEditToggle
+                shellCommand
+                dialogWarnings
+
+            }
+            Section {
+                updateChecker
+                autoUpdateToggle
+                prereleaseToggle
+            }
+        }
+    }
+}
+
+/// The extension of the view with all the preferences
+private extension GeneralSettingsView {
+    // MARK: - Sections
+
+    var appearance: some View {
+        Picker("Appearance", selection: $settings.general.appAppearance) {
+            Text("System")
+                .tag(SettingsData.Appearances.system)
+            Divider()
+            Text("Light")
+                .tag(SettingsData.Appearances.light)
+            Text("Dark")
+                .tag(SettingsData.Appearances.dark)
+        }
+        .onChange(of: settings.general.appAppearance) { tag in
+            tag.applyAppearance()
+        }
+    }
+
+    // TODO: Implement reflecting Show Issues preference and remove disabled modifier
+    var showIssues: some View {
+        Picker("Show Issues", selection: $settings.general.showIssues) {
+            Text("Show Inline")
+                .tag(SettingsData.Issues.inline)
+            Text("Show Minimized")
+                .tag(SettingsData.Issues.minimized)
+        }
+    }
+
+    var showLiveIssues: some View {
+        Toggle("Show Live Issues", isOn: $settings.general.showLiveIssues)
+    }
+
+    var fileExtensions: some View {
+        Group {
```

Comment:
This should be changed to a VStack to remove the separator between the selector and text view.

Reply:
See previous comment.